syntax = "proto2";

package gps;

// Enum for sample types
enum SampleType {
  ACTION = 0;
  CUR_LOC = 1;
  PAST_OBJ_VAL_DELTAS = 2;
  PAST_GRADS = 3;
  CUR_GRAD = 4;
  PAST_LOC_DELTAS = 5;
}

// Message containing the data for a single sample.
message Sample {
  optional uint32 T = 1;  // sample length
  optional uint32 dX = 2;  // dimension of state X
  optional uint32 dU = 3;  // dimension of action U
  optional uint32 dO = 4;  // dimension of observation

  // Data arrays holding X, U, obs, and meta data.
  repeated float X = 5 [packed = true];
  repeated float U = 6 [packed = true];
  repeated float obs = 7 [packed = true];
  repeated float meta = 8 [packed = true];
}
